packages/back-end/src/services/stats.ts

```diff
@@ -76,6 +76,7 @@ export interface MetricSettingsForStatsEngine {
   prior_proper?: boolean;
   prior_mean?: number;
   prior_stddev?: number;
+  decision_metric?: boolean; 
```

Comment:
Question for consideration. Should this instead be in AnalysisSettingsForStatsEngine as `bandit_metric?: string` where its an id. Then that why there's no chance of getting two decision metrics that return weights that compete? I'm actually not sure which is better, but just a thought.

Reply:
I accepted this change.  I also added test to ensure that for bandit analysis, the decision metric field is non-empty.

Reply:
Thanks for making the change. I'm not sure it does, because it still has to loop over a bunch of `DimensionResponse` objects to find the bandit weights. I think we could consider having it live at the `ExperimentMetricAnalysisResult` level so that it's trivially easy to find it.

I think we can resolve this later as well. Consider making the shit and if it's easy to manage with `gbstats.py` then go for it. Otherwise, maybe keep `BanditResponse` but remove it from `DimensionResponse` and we can figure out where it goes in the final integration PR.

Reply:
Thanks for your feedback.  As discussed offline, bandit_weights are now called in `process_experiment_results`.

Reply:
Thanks for pointing this out.  As discussed offline, I now call `get_bandit_weights` directly from `process_experiment_results`.